idaes_examples/mod/tut/visualizer_tutorial.py

```diff
@@ -289,7 +289,15 @@ def initialize_model(m: ConcreteModel) -> ConcreteModel:
     seq.set_guesses_for(m.fs.H101.inlet, tear_guesses)
 
     def initialize_unit(unit):
-        unit.initialize(outlvl=idaeslog.DEBUG)
+        try:
+            initializer = unit.default_initializer()
+            initializer.initialize(unit, output_level=idaeslog.INFO)
+        except Exception as e:
+            if e=='InitializationError':
+                initializer.initialize(unit, output_level=idaeslog.INFO)
+            else:
+                print(f'Unit {unit} did not initialize successfully')
+    
```

Comment:
Within the `except` block `e` will an instance of an `Exception` subclass, so it will never be evaluated equal to a string. From what I understand of the purpose of the code, one way to do this could be:

```suggestion
        try:
            initializer = unit.default_initializer()
            initializer.initialize(unit, output_level=idaeslog.INFO)
        except InitializationError:
            initializer.initialize(unit, output_level=idaeslog.INFO)
        except Exception:
            print(f'Unit {unit} did not initialize successfully')
    
```

Note that this requires `InitializationError` to be imported at the top of the file.

However, keep in mind that I'm completely unfamiliar with this code so this should be checked by someone with actual knowledge.

Reply:
Thank you. I shall check if that works. The above is the way I want to code to run but was not sure if I can write two except blocks with the try.

Reply:
@lbianchi-lbl Thank you this works.

Reply:
Just to check - is the intent of this that if the initialization fails the first time (with an InitializationError) to just try it again immediately afterwards? Yes, that is the intent. 

1. I agree, I should change the last exception and let the code terminate if the initialization was not successful even after the second solve. I shall also use the logger for the same. 
2. I am not sure if calling the solver was suffice. I shall check that. The reason the second step is necessary is because the code terminates if the initialization exceeds max iterations.

Reply:
For point 2, it is quite likely a scaling issue in which case calling a solver a gain has a decent chance of success. We could also look at changing the IPOPT settings - we can always increase max_iter, decrease the convergence tolerance (for initialization only) and also test changing bound_push and mu_init (again for initialization only).

Reply:
Yes, I tried with calling just the solver and it works. Thank you.